raphtory/src/serialise/serialise.rs

```diff
@@ -1,924 +1,162 @@
-                n_const_meta
-                    .ids()
-                    .flat_map(|i| node.constant_prop_layer(0, i).map(|v| (i, v))),
-            );
-        }
+            #[cfg(feature = "search")]
+            self.persist_index_to_disk_zip(&folder)?;
+        } else {
+            folder.reserve()?;
+            self.encode_parquet(&folder.get_graph_path())?;
 
-        // Edge Properties
-        let e_const_meta = self.edge_meta().metadata_mapper();
-        for (id, key, dtype) in e_const_meta.locked().iter_ids_and_types() {
-            graph.new_edge_cprop(key, id, dtype);
-        }
-        let e_temporal_meta = self.edge_meta().temporal_prop_mapper();
-        for (id, key, dtype) in e_temporal_meta.locked().iter_ids_and_types() {
-            graph.new_edge_tprop(key, id, dtype);
+            #[cfg(feature = "search")]
+            self.persist_index_to_disk(&folder)?;
         }
 
-        // Edges
-        let edges = storage.edges();
-        for edge in edges.iter(&LayerIds::All) {
-            let eid = edge.eid();
-            let edge = edge.as_ref();
-            graph.new_edge(edge.src(), edge.dst(), eid);
-            for layer_id in storage.unfiltered_layer_ids() {
-                for (t, props) in zip_tprop_updates!(e_temporal_meta
-                    .ids()
-                    .map(|i| (i, edge.temporal_prop_layer(layer_id, i))))
-                {
-                    graph.update_edge_tprops(eid, t, layer_id, props.map(|(_, v)| v));
-                }
-                for t in edge.additions(layer_id).iter() {
-                    graph.update_edge_tprops(eid, t, layer_id, iter::empty::<(usize, Prop)>());
-                }
-                for t in edge.deletions(layer_id).iter() {
-                    graph.del_edge(eid, layer_id, t);
-                }
-                graph.update_edge_cprops(
-                    eid,
-                    layer_id,
-                    e_const_meta
-                        .ids()
-                        .filter_map(|i| edge.metadata_layer(layer_id, i).map(|prop| (i, prop))),
-                );
-            }
-        }
-        graph
-    }
-}
+        folder.write_metadata(self)?;
 
-impl StableEncode for Graph {
-    fn encode_to_proto(&self) -> proto::Graph {
-        let mut graph = self.core_graph().encode_to_proto();
-        graph.set_graph_type(proto::GraphType::Event);
-        graph
+        Ok(())
     }
 }
 
-impl StableEncode for PersistentGraph {
-    fn encode_to_proto(&self) -> proto::Graph {
-        let mut graph = self.core_graph().encode_to_proto();
-        graph.set_graph_type(proto::GraphType::Persistent);
-        graph
-    }
-}
+pub trait StableDecode: StaticGraphViewOps + AdditionOps {
+    // Decode the graph from the given bytes array.
+    // `path_for_decoded_graph` gets passed to the newly created graph.
+    fn decode_from_bytes(bytes: &[u8], path_for_decoded_graph: Option<impl AsRef<Path>>) -> Result<Self, GraphError>;
```

Comment:
make the second input just `Option<&Path>` to make it less annoying to pass None (and only mildly more annoying to pass `Some(path)`

Reply:
done